extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/S3TimestampVersionedDataFinder.java

```diff
@@ -44,13 +44,10 @@ public S3TimestampVersionedDataFinder(RestS3Service s3Client)
   }
 
   /**
-   * Gets the latest version using the "parent" of uri as a prefix. The "parent" of uri is evaluated assuming '/'
-   * delimited paths. If the uri path ends with '/', the path is assumed to be the parent.
+   * Gets the key with the most recently modified timestamp. `pattern` is evaluated against the entire key.
```

Comment:
The impl looks to be evaluating pattern against the part of the key after "uri" (and possibly one more "/"), not the entire key. imo what the impl is doing is reasonable (and in line with the other finders) so would just change this doc to something like,

`pattern` is evaluated against the portion of the key following `uri`. If that portion of the key begins with a "/" character, it will be stripped before applying the `pattern`. Only one slash will be stripped in this manner.